### PR TITLE
Handle Bellatrix Fork Epochs Correctly

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Fork Addition

**What does this PR do? Why is it needed?**

- [x] It updates prysm's fork watcher to handle bellatrix forks correctly. Instead of only handling altair, we now handle for
bellatrix as well.
- [x] Add in relevant test cases .

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/sync/fork_watcher.go [hunk 1]
```diff
@@ -59,6 +60,15 @@ func (s *Service) registerForUpcomingFork(currEpoch types.Epoch) error {
 			}
 			s.registerSubscribers(nextEpoch, digest)
 			s.registerRPCHandlersAltair()
+		case params.BeaconConfig().BellatrixForkEpoch:
+			digest, err := forks.ForkDigestFromEpoch(nextEpoch, genRoot[:])
+			if err != nil {
+				return errors.Wrap(err, "Could not retrieve fork digest")
```
Comment (rkapka) [inline beacon-chain/sync/fork_watcher.go:66]:
```suggestion
				return errors.Wrap(err, "could not retrieve fork digest")
```

**Review by rkapka — CHANGES_REQUESTED** [review]

File: beacon-chain/sync/fork_watcher.go [hunk 2]
```diff
@@ -95,7 +105,15 @@ func (s *Service) deregisterFromPastFork(currEpoch types.Epoch) error {
...
-		s.unregisterPhase0Handlers()
+		prevFork, err := forks.Fork(epochBeforeFork)
+		if err != nil {
+			return errors.Wrap(err, "Failed to determine previous epoch fork data")
+		}
+
+		switch prevFork.Epoch {
+		case params.BeaconConfig().GenesisEpoch:
+			s.unregisterPhase0Handlers()
+		}
```
Comment (rkapka) [inline beacon-chain/sync/fork_watcher.go:116]: Switch with a single case? I presume this is a placeholder for Altair unregistrations?

Comment (nisdas) [reply beacon-chain/sync/fork_watcher.go:116]: There are no altair unregistrations, rpc methods stay the same in bellatrix

Comment (rkapka) [reply beacon-chain/sync/fork_watcher.go:116]: Isn't it more natural to write
```
if prevFork.Epoch == params.BeaconConfig().GenesisEpoch {
    s.unregisterPhase0Handlers()
}
```

Comment (nisdas) [reply beacon-chain/sync/fork_watcher.go:116]: We will have new methods most likely added/removed in newer forks. The switch brings it inline with fork handling code across prysm.

**Review by rkapka — APPROVED** [review]

**Review by terencechain — APPROVED** [review]

**Pull request merged**